### ISSUE TITLE
Can't Return to ChromeOS

### ISSUE THREAD

**Issue by angryvegetables** [dnschneid/crouton#1935]
When I start my Chromebook and launch crouton Linux starts fine. However, if I press CTRL+ALT+SHIFT+<== to return to chrome it gives me a black screen. If I press CTRL+ALT+SHIFT+==> I can return to Linux, but I can't use chrome until I restart my computer.

Here's the output of cat /etc/lsb-release
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={A772AA64-E906-A01E-1DFD-1856870D39EC}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={A772AA64-E906-A01E-1DFD-1856870D39EC}
CHROMEOS_RELEASE_BOARD=quawks-signed-mp-v2keys
CHROMEOS_RELEASE_BRANCH_NUMBER=70
CHROMEOS_RELEASE_BUILD_NUMBER=6946
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=43
CHROMEOS_RELEASE_DESCRIPTION=6946.70.0 (Official Build) stable-channel quawks 
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_VERSION=6946.70.0
GOOGLE_RELEASE=6946.70.0

Thanks

**Comment by angryvegetables** [comment]
Here's the output of croutonversion

crouton: version 1-20150604154515~master:91b3e350
release: trusty
architecture: amd64
xmethod: xorg
targets: xfce
host: version 6946.70.0 (Official Build) stable-channel quawks 
kernel: Linux localhost 3.10.18 #1 SMP Tue Jul 7 19:31:09 PDT 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: yes

**Comment by DennisLfromGA** [comment]
While in your chroot, what happens when you enter `croutoncycle cros` in a shell?

**Comment by angryvegetables** [comment]
It takes me to a black screen like if I pressed CTRL+ALT+SHIT+<== to return to chrome. I can still return to linux from the black screen with CTRL+ALT+SHIFT+==>

**Comment by DennisLfromGA** [comment]
Bummer. 
You could try the 'xiwi' xmethod, it's easy to switch in and out of without using keyboard shortcuts but it doesn't have hardware acceleration at the moment.
You can add it via a crouton update like: `sudo sh ~/Downloads/crouton -t xiwi -n trusty -u`
It'll become the default xmethod so if you want to use 'xorg' again you have to specify '-X xorg' in your 'start*' script command(s).

**Comment by angryvegetables** [comment]
When I input that I get 

/usr/local/chroots not found
/usr/local/chroots/trusty does not exist; cannot update.
Valid Chroots:

Does this mean I need to repair or reinstall crouton

**Comment by DennisLfromGA** [comment]
Well, you had a trusty chroot at one time, the 'croutonversion' output above showed that.
Perhaps it's not in `/usr/local/chroots`, is it on external media or someplace else?

> Valid Chroots:
> 
> Does this mean I need to repair or reinstall crouton

The blank line following 'Valid Chroots:' does not look promising, did you delete your chroot somehow?

You could try the command without '-n trusty' but I doubt if that'd help at this point... :/

**Comment by angryvegetables** [comment]
Here's what happened.

(precise)admin@localhost:~$ sudo sh ~/Downloads/crouton -t xiwi -u           
/usr/local/chroots/precise already exists; updating it...
WARNING: Your rootfs is writable. Signed boot verification cannot be enabled.
If this is a surprise to you, you should do a full system recovery via USB.
Preparing chroot environment...
Installing brightness into the chroot...
Installing croutonpowerd into the chroot...
Installing croutonversion into the chroot...
Installing host-dbus into the chroot...
Installing host-x11 into the chroot...
Installing volume into the chroot...
Installing pulseaudio-default.pa into the chroot...
Installing croutonclip into the chroot...
Installing croutonnotify into the chroot...
Installing croutonurlhandler into the chroot...
Installing startxiwi into the host...
Installing croutoncycle into the chroot...
Installing croutonfindnacl into the chroot...
Installing croutontriggerd into the chroot...
Installing croutonxinitrc-wrapper into the chroot...
Installing setres into the chroot...
Installing xinit into the chroot...
Installing xiwi into the chroot...
Installing xbindkeysrc.scm into the chroot...
Installing xiwi.conf into the chroot...
Installing xorg-dummy.conf into the chroot...
Installing xserverrc into the chroot...
Installing xserverrc-xiwi into the chroot...
Installing xserverrc-local.example into the chroot...
Installing startxfce4 into the host...
Installing crouton-noroot into the chroot...
Installing startxfce4 into the chroot...
Installing enter-chroot into the host...
Installing delete-chroot into the host...
Installing edit-chroot into the host...
Installing mount-chroot into the host...
Installing unmount-chroot into the host...
Installing crash_reporter_wrapper into the host...
mount: Too many levels of symbolic links
WARNING: Unable to register core dump handler.
mount: special device /var/run/shill does not exist
Unmounting /usr/local/chroots/precise...

**Comment by DennisLfromGA** [comment]
> (precise)admin@localhost:~$ sudo sh ~/Downloads/crouton -t xiwi -u

You're attempting to update crouton from within the chroot, it should be done in a Chrome OS shell, outside of your chroot.

**Comment by angryvegetables** [comment]
I got xiwi to work but it makes stuff on the linux side run extremely slow. Is there anything I can do to make xorg allow me to switch back

**Comment by darcyparker** [comment]
I am having the same issue.  I think it may be related to this issue I reported awhile back for ChromeOS: https://code.google.com/p/chromium/issues/detail?id=475330

As recommend, I got xiwi to work.  I haven't tested significantly, but it seems to work okay for a terminal.   But xiwi's performance isn't great for other graphical apps. I'd really like to get this working like it used to where I could switch back and forth between the VT for the chroot and ChromeOS.
